File: src/test/java/trilby/test/GenHeap.java
package trilby.test;

import java.util.Map;

import com.google.common.collect.Maps;

public class GenHeap
{
    private Map<Integer,Object> m = Maps.newHashMap();
    
    {
        class Thing1 {int x;};
        
        for (int i = 0; i < 10; i++)
            m.put(i, new Thing1());
        
        /*
        for (int i = 0; i < 10000; i++) {
            m.put(i, String.valueOf(i));
        }
        */
    }
    
    public static void main(String[] args) throws Exception {
        GenHeap g = new GenHeap();
        Thread.sleep(60000);
    }
}

File: src/main/scala/trilby/main/GenHeap.scala
package trilby.main
import java.util.Random
import java.util.ArrayList
import java.util.HashMap

object GenHeap {
    
    val map = new HashMap[Int,ArrayList[String]]
    
    def main(args: Array[String]) {
        
        val count = args(0).toInt
        val rand = new Random()
        var list = new ArrayList[String]()
        
        for (i <- 1 to count) {
            list.add(String.valueOf(i))
            if (rand.nextInt(10) == 0) {
                map.put(i, list)
                list = new ArrayList[String]()
            }
        }
        
        println("Sleeping")
        Thread.sleep(60 * 1000)
    }

}
